parser.go

```diff
@@ -35,3 +37,55 @@ func (p *ErrorformatParser) Parse(r io.Reader) ([]*CheckResult, error) {
 	}
 	return rs, nil
 }
+
+var _ Parser = &CheckStyleParser{}
+
+// CheckStyleParser is checkstyle parser.
+type CheckStyleParser struct{}
+
+// NewCheckStyleParser returns a new CheckStyleParser.
+func NewCheckStyleParser() Parser {
+	return &CheckStyleParser{}
+}
+
+func (p *CheckStyleParser) Parse(r io.Reader) ([]*CheckResult, error) {
+	var cs = new(CheckStyleResult)
+	if err := xml.NewDecoder(r).Decode(cs); err != nil {
+		return nil, err
+	}
+	var rs []*CheckResult
+	for _, file := range cs.Files {
+		for _, cerr := range file.Errors {
+			rs = append(rs, &CheckResult{
+				Path:    file.Name,
+				Lnum:    cerr.Line,
+				Col:     cerr.Column,
+				Message: cerr.Message,
+				Lines: []string{
+					fmt.Sprintf("%v:%d:%d: %v: %v (%v)",
+						file.Name, cerr.Line, cerr.Column, cerr.Severity, cerr.Message, cerr.Source),
+				},
+			})
+		}
+	}
+	return rs, nil
+}
+
+type CheckStyleResult struct {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type CheckStyleResult should have comment or be unexported